app/src/main/java/com/chess/clock/engine/TimeControlParser.java

```diff
@@ -189,6 +193,14 @@ public static int getLastTimeControlCheckIndex(Context context) {
             Log.w(TAG, "Not able to read the preference");
             return null;
         }
+        //migrate old format to new
+        boolean userHasOldVersion = jsonString.contains(TC_JSON_TIME_CONTROLS_OLD_VERSION);
```

Comment:
Please add `@SuppressWarnings("deprecation")` to this method, as using those constants is justified in this case.
Or maybe extract migration to private static method and put `timecontrols` `timeincrement` as local variables inside it, as it is unlikely that we will want to use them anywhere else.